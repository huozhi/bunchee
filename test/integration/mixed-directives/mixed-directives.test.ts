import { readFileSync } from 'fs'
import { createIntegrationTest, getFileNamesFromDirectory } from 'testing-utils'

describe('integration - mixed-directives', () => {
  it('should work with js only project', async () => {
    await createIntegrationTest(
      {
        directory: __dirname,
      },
      async ({ distDir }) => {
        const distFiles = await getFileNamesFromDirectory(distDir)
        const fileContents = distFiles
          .map((file) => [file, readFileSync(`${distDir}/${file}`, 'utf-8')])
          .reduce((acc, pair) => {
            return {
              ...acc,
              [pair[0]]: pair[1],
            }
          }, {})

        expect(fileContents).toMatchSnapshot(``)
      },
    )
  })
})
