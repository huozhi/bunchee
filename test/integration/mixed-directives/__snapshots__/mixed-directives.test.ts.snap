// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration - mixed-directives should work with js only project 1`] = `
{
  "action-server-DOTFC6td.js": "'use server';
async function action1() {
    return 'action1';
}

export { action1 as a };
",
  "client.js": "'use client';
import { jsx } from 'react/jsx-runtime';
import { a as action1 } from './action-server-DOTFC6td.js';

function Page() {
    return /*#__PURE__*/ jsx("button", {
        onClick: action1,
        children: "Action 1"
    });
}

export { Page as default };
",
  "inline.js": "import { jsx } from 'react/jsx-runtime';
import ClientComponent from 'client-component';

// @ts-ignore externals
function Page() {
    async function inlineAction() {
        'use server';
        return 'inline-action';
    }
    return /*#__PURE__*/ jsx(ClientComponent, {
        action: inlineAction
    });
}

export { Page as default };
",
}
`;
